jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -5466,12 +5500,96 @@ public Object helper(Object[] arguments) throws Throwable {
 	}
 
 	/*[IF Java15]*/
-	static boolean permuteArgumentChecks(int[] arr, MethodType mt1, MethodType mt2) {
-		throw OpenJDKCompileStub.OpenJDKCompileStubThrowError();
+	/**
+	 * Validates that the permute[] specifies a valid permutation from permuteType to handleType.
+	 * This method throws IllegalArgumentException on failure and returns true on success. This
+	 * disjointness allows it to be used in asserts.
+	 * 
+	 * @param permute array specifies the conversion from permuteType to handleType.
+	 * @param permuteType source method type.
+	 * @param handleType target method type.
+	 * 
+	 * @return true on success.
+	 * @throws IllegalArgumentException on failure.
+	 */
+	static boolean permuteArgumentChecks(int[] permute, MethodType permuteType, MethodType handleType) {
+		return validatePermutationArray(permuteType, handleType, permute);
 	}
 	
-	static MethodHandle collectReturnValue(MethodHandle mh1, MethodHandle mh2) {
-		throw OpenJDKCompileStub.OpenJDKCompileStubThrowError();
+	/**
+	 * Return the classData stored in the accessClass of the Lookup object.
+	 * 
+	 * @param caller Lookup object used to verify privileged access and retrieve classData.
+	 * @param unused.
```

Comment:
Why an unused parameter?

Reply:
RI introduced `MethodHandles.classData` with an `unused String`. We had to match the RI in order to support calls from the JEP383 API.